src/main/java/frc/robot/subsystems/endEffector/TOFSensor.java

```diff
@@ -0,0 +1,76 @@
+package frc.robot.subsystems.endEffector;
+
+import com.playingwithfusion.TimeOfFlight;
+import com.playingwithfusion.TimeOfFlight.RangingMode;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class TOFSensor {
+  public double m_distance_EMA;
+  public double m_dist_SDEV_sq;
+  public String mode = "Short";
+
+  int m_rangeX0;
+  int m_rangeY0;
+  int m_rangeX1;
+  int m_rangeY1;
+
+  private TimeOfFlight sensor;
+
+  public TOFSensor(int CAN_ID) {
+    sensor = new TimeOfFlight(CAN_ID);
+    RoI(0, 0, 15, 15);
+    sensor.setRangingMode(RangingMode.Short, 0.24); // Will need to test. must be between 24-1000ms
+    m_distance_EMA = sensor.getRange();
+    m_dist_SDEV_sq = 0;
+  }
+
+  public void RoI(int rangeX0, int rangeY0, int rangeX1, int rangeY1) {
+    m_rangeX0 = rangeX0;
+    m_rangeY0 = rangeY0;
+    m_rangeX1 = rangeX1;
+    m_rangeY1 = rangeY1;
+
+    sensor.setRangeOfInterest(m_rangeX0, m_rangeY0, m_rangeX1, m_rangeY1);
+    SmartDashboard.putNumber("rangeX0", m_rangeX0);
+    SmartDashboard.putNumber("rangeY0", m_rangeY0);
+    SmartDashboard.putNumber("rangeX1", m_rangeX1);
+    SmartDashboard.putNumber("rangeY1", m_rangeY1);
+  }
+
+  public double getRange() {
+    return sensor.getRange();
+  }
+
+  public void robotPeriodic() {
+
+    // /*for (int topLeftRow = 0; topLeftRow < 12; topLeftRow++) {
+    //   System.out.print("previous row");
+    //   System.out.print(topLeftRow);
+    //   for (int topLeftColumn = 0; topLeftColumn < 12; topLeftColumn++) {
+    //     sensor.setRangeOfInterest(topLeftColumn, topLeftRow, topLeftColumn + 4, topLeftRow + 4);
+    //     System.out.print(sensor.getRange() + ",");
+    //     try {
+    //       Thread.sleep(1000);
+    //     } catch (InterruptedException e) {
+    //       // TODO Auto-generated catch block
+    //       e.printStackTrace();
+    //     }
+    //     System.out.print(sensor.getStatus());
+    //   }
+    //   System.out.println();
+    // }*/
+
+    double distance = sensor.getRange();
+    SmartDashboard.putNumber("Distance", distance);
+
+    m_dist_SDEV_sq =
+        (EndEffectorConstants.SDEV_Decay * Math.pow((m_distance_EMA - sensor.getRange()), 2))
```

Comment:
Math.pow is inefficient if you are just squaring a number.  Just subtract the values, put in a local variable, and use multiply

Reply:
That's so tragic that math.pow isn't optimized for this.